src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/CourtVenueServiceImpl.java

```diff
@@ -205,4 +228,81 @@ private void handleIfCourtVenuesEmpty(BooleanSupplier courtVenueResponseVerifier
             throw new ResourceNotFoundException(noDataFoundMessage);
         }
     }
+
+    private List<LrdCourtVenueResponse> getLrdCourtVenueResponses(
+
+        List<LrdCourtVenueResponse> inputLrdCourtVenueResponse,
+        CourtVenueRequestParam courtVenueRequestParam) {
+
+        List<LrdCourtVenueResponse> result = applyAdditionalFilters(
+            inputLrdCourtVenueResponse,
+            courtVenueRequestParam
+        );
+        if (isEmpty(result)) {
+            throw new ResourceNotFoundException(NO_COURT_VENUES_FOUND);
+        }
+        return result;
+    }
+
+    private List<LrdCourtVenueResponse> applyAdditionalFilters(
+
+        List<LrdCourtVenueResponse> inputLrdCourtVenueResponse,
+        CourtVenueRequestParam courtVenueRequestParam) {
+
+        List<Predicate<LrdCourtVenueResponse>> allPredicates = getPredicates(courtVenueRequestParam);
+
+
+        return inputLrdCourtVenueResponse.stream()
+            .filter(allPredicates.stream().reduce(x -> true, Predicate::and))
+            .collect(Collectors.toList());
+
+    }
+
+    private List<Predicate<LrdCourtVenueResponse>> getPredicates(
+
+        CourtVenueRequestParam courtVenueRequestParam) {
+
+        List<Predicate<LrdCourtVenueResponse>> allPredicates =
+            new ArrayList<>();
+
+        String isHearingLocationValue = courtVenueRequestParam.getIsHearingLocation();
+
+        if (IS_HEARING_LOCATION_Y.equalsIgnoreCase(isHearingLocationValue)
+            || IS_HEARING_LOCATION_N.equalsIgnoreCase(isHearingLocationValue)) {
+            allPredicates.add(
+                courtVenue -> isHearingLocationValue.equalsIgnoreCase(courtVenue.getIsHearingLocation())
+            );
+
+        }
+
+        String isCaseMgntLocationValue = courtVenueRequestParam.getIsCaseManagementLocation();
+
+        if (IS_CASE_MANAGEMENT_LOCATION_Y.equalsIgnoreCase(isCaseMgntLocationValue)
```

Comment:
Can't we just check isCaseMgntLocationValue != null? as the validation for Y or N is already done